Specified minimum gradle version required

sergico opened the pull request:
OS: Ubuntu 14.04 64b
Trying to build a sample VNFM
Following the doc I got the following error:

openbaton@openbaton-VBox01:~/develop/openbaton/nr/simple-vnfm$ gradle wrapper --gradle-version 2.4

FAILURE: Build failed with an exception.
- What went wrong:
  Could not resolve all dependencies for configuration ':classpath'.
  > Could not find org.springframework.boot:spring-boot-gradle-plugin:1.2.6.RELEASE.
  Required by:
      :simple-vnfm:unspecified

Looking at the spring-boot documentation it seems that the minimum required gradle version is 1.12:
http://docs.spring.io/spring-boot/docs/current-SNAPSHOT/reference/htmlsingle/#getting-started-system-requirements
Ubuntu 14.04 gradle default version is (apt-get) is 1.4
Updated doc accordingly

spring-boot documentation also suggest maven 3.2
Shall we add this dependency as well to the doc?

sergico committed:
Specified minimum gradle version required

sergico committed:
Modified gradle build.gradle script

lorenzotomasini commented:
Hi,

the gradle version installed from apt is 1.0-milestone-3, so as you wrote, we need to specify the gradle version requirement to 1.2 in the doc. 

Moreover there is the need of adding the repo in the buildscript scope of the build gradle file. 

But I don't understand the need of adding the milestone and snapshot repos, and also the other lines such as jar scope, dependencies and the reason why you removed group and version. Anyway the details you added are not depending on OpenBaton, so i think is not the scope of this documentation.

for me the final version would be:

``` gradle
buildscript {
    repositories{
        mavenCentral()
    }
    dependencies {
        classpath("org.springframework.boot:spring-boot-gradle-plugin:1.3.1.RELEASE")
    }
}
apply plugin: 'spring-boot'
apply plugin: 'java'

group "your.group"
version '1.0-SNAPSHOT'
```

Any comment from your side on this?

thanks,
Lorenzo

sergico committed:
Fixed the build.gradle script according to github discussion
- removed unnecessary repos
- removed unnecessary scopes

sergico commented:
Hi Lorenzo,

this is the first time I work with gradle, I do not have enough knowledge on the topic to provide a reasonable argument. 
What I was trying to achieve was to create the wrapper, and I noticed the current script was not working for me. I basically modified the script looking at the examples in the sprint-boot documentation.
The new version of the script seems reasonable to me, meaning that:
1) I agree, that the snapshot and milestone repos are not really required.
2) you removed the **maven** plugin, that was present in the previous version of the script. Do you think it is not needed anymore?

If this is ok with you I will send a new pull request to the doc with the new script version, as proposed from you.

sergico committed:
updated spring-boot-gradle-plugin version to 1.3.1

sergico committed:
removed slash (path is not absolute to root)

lorenzotomasini commented:
Yes,

maven is not needed, not for a creating a simple vnfm so we can remove it from the build.gradle.

sergico committed:
updated overall build.gradle script

sergico committed:
Fixed mainClass name in bootRepackage scope

sergico committed:
- updated vnfm skeleton to match the latest interface
- updated the setup() sample code to match teh new startPluginRecursive interface
  - added comments trying to explain the meaning of teh different params

sergico committed:
- converted tabs -> spaces to display formatted code correctly

sergico committed:
- fixed typo in curl command
